src/mozanalysis/frequentist_stats/linear_models/functions.py

```diff
@@ -334,27 +369,58 @@ def fit_model(
 
     Returns:
     - results (RegressionResults): the fitted model results object.
+
+    Raises:
+    - `FailedToFitModel` if unable to fit the model.
+    - `Exception` if certain unexpected conditions arise (e.g., one of the treatment)
+    branches, which was expected to have a model parameter, does not get one.
```

Comment:
Typo, but also -- should we have a more specific Exception for this case? I think it is usually preferred to raise a more descriptive exception type whenever possible (this makes it easier for exception handling because it allows the calling code to catch a specific exception instead of having to catch `Exception` which would be anything). It doesn't necessarily have to be a custom Exception type that we create, but that just depends on what's happening.

```suggestion
    - `Exception` if certain unexpected conditions arise (e.g., one of the treatment
    branches, which was expected to have a model parameter, does not get one).
```

Reply:
Done